pyanaconda/payload/dnf/payload.py

```diff
@@ -204,11 +204,39 @@ def set_packages_selection(self, data: PackagesSelectionData):
 
     def is_ready(self):
         """Is the payload ready?"""
-        return self.base_repo is not None
+        enabled_repos = self._dnf_manager.enabled_repositories
+
+        # If CDN is used as the installation source and we have
+        # a subscription attached then any of the enabled repos
+        # should be fine as the base repo.
+        # If CDN is used but subscription has not been attached
+        # there will be no redhat.repo file to parse and we
+        # don't need to do anything.
+        if self.source_type == SOURCE_TYPE_CDN:
+            return self._is_cdn_set_up() and enabled_repos
+
+        # Otherwise, a base repository has to be enabled.
+        return any(map(self._is_base_repo, enabled_repos))
+
+    def _is_cdn_set_up(self):
```

Comment:
@M4rtinK, could you review these changes, please?

Reply:
Yep, this shduld work correctly as far as I can tell from looking at the code.

Unfortunately we can't really test this in a realistic scenario on Fedora just yet, though I have some ideas in the longer term. :)